### DISCUSSION TITLE
Welcome to NLPModelsIpopt.jl Discussions!

### DISCUSSION THREAD

**Discussion by dpo** [JuliaSmoothOptimizers/NLPModelsIpopt.jl#99] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by jbcaillau** [comment]
Hi @amontoison and @tmigot ; regarding [these tips](https://jso.dev/NLPModelsIpopt.jl/previews/PR113/tips/), any thoughts on [LinearSolve.jl](https://docs.sciml.ai/LinearSolve/stable)?

**Reply by amontoison** [reply]
@jbcaillau Do you need to solve linear systems explicitly?

**Reply by jbcaillau** [reply]
@amontoison Indeed, currently not. Just wondering whether the proposed type of interface was relevant.

**Reply by amontoison** [reply]
The biggest issue of these packages (LinearSolver.jl, Optimization.jl, DiffEq.jl, ...) is the number of dependencies.
When you add it to you package, you add an heavy dependency that your users probably don't want.

From my point of view, it should be used only for prototyping. When you find the best solver for you, you interface it in your package.
You will have more options and highly reduce the precompilation time.
For example LinearSolve.jl contains Krylov.jl but many features and solvers are not supported.

**Reply by jbcaillau** [reply]
@amontoison agreed. actually, this is exactly the kind of things we need to look at for `OptimalControl.jl` that has huge precompilation times.
